internal/provider/resource_cloudflare_waiting_room_event.go

```diff
@@ -42,19 +42,21 @@ func expandWaitingRoomEvent(d *schema.ResourceData) (cloudflare.WaitingRoomEvent
 		return cloudflare.WaitingRoomEvent{}, err
 	}
 
-	prequeueStartTime := time.Time{}
+	var prequeueStartTime *time.Time
 	if t, ok := d.GetOk("prequeue_start_time"); ok {
-		prequeueStartTime, err = time.Parse(time.RFC3339, t.(string))
+		prequeueStartTimeValue, err := time.Parse(time.RFC3339, t.(string))
```

Comment:
we have `Time` and `TimePtr` if you'd like to use those here instead to manually go back and forth.

https://github.com/cloudflare/cloudflare-go/blob/master/convert_types.go#L67-L69

https://github.com/cloudflare/cloudflare-go/blob/master/convert_types.go#L916-L923